Source/Core/Core/PowerPC/GDBStub.cpp

```diff
@@ -305,23 +521,90 @@ static void SendReply(const char* reply)
   }
 }
 
+static void ProcessXFerCommand(const char* data, size_t paramsIndex)
+{
+  size_t offset = 0;
+  size_t length = 0;
+
+  while (s_cmd_bfr[paramsIndex] != ',')
+  {
+    offset <<= 4;
+    offset |= Hex2char(s_cmd_bfr[paramsIndex]);
+    paramsIndex++;
+  }
+
+  paramsIndex++;
+  while (paramsIndex < s_cmd_len)
+  {
+    length <<= 4;
+    length |= Hex2char(s_cmd_bfr[paramsIndex]);
+    paramsIndex++;
+  }
+  static u8 reply[GDB_BFR_MAX];
+  memset(reply, 0, GDB_BFR_MAX);
+  if (length + 1 > GDB_BFR_MAX)
+  {
+    memcpy(reply, "E00", 3);
+    return SendReply(reinterpret_cast<const char*>(reply));
+  }
+
+  if (strlen(data) < static_cast<size_t>(length) + offset)
+  {
+    length = strlen(data) - offset;
+    reply[0] = 'l';
+  }
+  else
+  {
+    reply[0] = 'm';
+  }
+  memcpy(&reply[1], &data[offset], length);
+  return SendReply(reinterpret_cast<const char*>(reply));
+}
+
+static bool IsQuery(const char* query)
+{
+  return !strncmp((const char*)(s_cmd_bfr), query, strlen(query));
+}
+
 static void HandleQuery()
 {
   DEBUG_LOG_FMT(GDB_STUB, "gdb: query '{}'", CommandBufferAsString());
 
-  if (!strncmp((const char*)(s_cmd_bfr), "qAttached", strlen("qAttached")))
+  if (IsQuery("qAttached"))
+  {
     return SendReply("1");
-  if (!strcmp((const char*)(s_cmd_bfr), "qC"))
+  }
+  else if (IsQuery("qC"))
+  {
     return SendReply("QC1");
-  if (!strcmp((const char*)(s_cmd_bfr), "qfThreadInfo"))
+  }
+  else if (IsQuery("qfThreadInfo"))
+  {
     return SendReply("m1");
-  else if (!strcmp((const char*)(s_cmd_bfr), "qsThreadInfo"))
+  }
+  else if (IsQuery("qsThreadInfo"))
+  {
     return SendReply("l");
-  else if (!strncmp((const char*)(s_cmd_bfr), "qThreadExtraInfo", strlen("qThreadExtraInfo")))
+  }
+  else if (IsQuery("qThreadExtraInfo"))
+  {
     return SendReply("00");
-  else if (!strncmp((const char*)(s_cmd_bfr), "qSupported", strlen("qSupported")))
-    return SendReply("swbreak+;hwbreak+");
-
+  }
+  else if (IsQuery("qSupported"))
+  {
+    return SendReply("swbreak+;hwbreak+;qXfer:features:read+;qXfer:memory-map:read+");
+  }
+  else if (IsQuery("qXfer:features:read:target.xml:"))
+  {
+    ProcessXFerCommand(target_xml, strlen("qXfer:features:read:target.xml:"));
+  }
+  else if (IsQuery("qXfer:memory-map:read::"))
+  {
+    const char* memoryMapXml = target_memory_map_xml_noMMU;
+    if (SConfig::GetInstance().bMMU)
+      memoryMapXml = target_memory_map_xml_withMMU;
+    ProcessXFerCommand(memoryMapXml, strlen("qXfer:memory-map:read::"));
```

Comment:
Should the query commands be constants? `qXfer:memory-map:read::` is a rather complicated string and used twice, which may or may not be a problem in the future. (easy to forget changing one but not the other for example).

Reply:
We could just constantize the xfer ones, they are especially long and checking other queries, it's probably the longest we'll deal with. Is that ok?

Reply:
Works for me!